addon/MetadataParser.js

```diff
@@ -41,7 +41,7 @@ MetadataParser.prototype = {
     return new Promise(resolve => {
       const doc = this._getDocumentObject(raw);
       resolve(this._formatData(getMetadata(doc, url), url));
-    });
+    }).catch(err => Cu.reportError(err));
```

Comment:
If you handle the error here, the error handling in `_asyncParseAndSave` in `PageScraper.js` will never get hit